buildkitd/certificates.go

```diff
@@ -247,3 +250,60 @@ func savePEM(path, typ string, bytes []byte) error {
 
 	return nil
 }
+
+// ConfigureSatelliteTLS uses the CA cert and key associate with the satellite
+// to generate a new certificate/key pair for use in client-side mTLS.
+// The certificates are configured in the settings for a new buildkit client.
+func ConfigureSatelliteTLS(settings *Settings, sat *cloud.SatelliteInstance) (cleanupFn func(), err error) {
+	dir := filepath.Join(os.TempDir(), "earthly", "certs", uuid.NewString())
+	caRootPath := filepath.Join(dir, "root_ca_cert.pem")
+	caCertPath := filepath.Join(dir, "ca_cert.pem")
+	caKeyPath := filepath.Join(dir, "ca_key.pem")
+	earthlyCertPath := filepath.Join(dir, "earthly_cert.pem")
+	earthlyKeyPath := filepath.Join(dir, "earthly_key.pem")
+
+	settings.TemporaryTLS = true
+	settings.TLSDir = dir
+	settings.TLSCA = caRootPath
+	settings.ClientTLSCert = earthlyCertPath
+	settings.ClientTLSKey = earthlyKeyPath
+	settings.ServerTLSCert = ""
+	settings.ServerTLSKey = ""
+
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return nil, err
+	}
```

Comment:
after this point there's a leak where the dir isn't cleaned up if any errors are returned.

maybe something like:

```
cleanup := func() { _ = os.RemoveAll(dir) }
defer func() {
    if err != nil {
        cleanup()
    }
}

...

return cleanup, nil // the final return in this function
```

Reply:
Good point. I overlooked that.